src/app/clusters/operational-credentials-server/operational-credentials-server.cpp

```diff
@@ -991,33 +1031,43 @@ bool emberAfOperationalCredentialsClusterCSRRequestCallback(app::CommandHandler
 #else
         keypair.Initialize();
 #endif
-        SuccessOrExit(err = gFabricBeingCommissioned.SetOperationalKeypair(&keypair));
+        err = gFabricBeingCommissioned.SetOperationalKeypair(&keypair);
+        VerifyOrExit(err == CHIP_NO_ERROR, finalStatus = Status::Failure);
 
         // Generate the actual CSR from the ephemeral key
-        VerifyOrExit(csr.Alloc(csrLength), err = CHIP_ERROR_NO_MEMORY);
+        if (!csr.Alloc(csrLength))
+        {
+            err = CHIP_ERROR_NO_MEMORY;
+            VerifyOrExit(err == CHIP_NO_ERROR, finalStatus = Status::Failure);
```

Comment:
Followup: This should be Status::ResourceExhausted.

Reply:
I agree that it's not clear what a client is to do here in the face of this error...